homeassistant/components/cover/__init__.py

```diff
@@ -333,10 +346,18 @@ def toggle_tilt(self, **kwargs: Any) -> None:
 
     async def async_toggle_tilt(self, **kwargs):
         """Toggle the entity."""
-        if self.current_cover_tilt_position == 0:
-            await self.async_open_cover_tilt(**kwargs)
+        await self.hass.async_add_executor_job(ft.partial(self.toggle_tilt, **kwargs))
+
+    def async_write_ha_state(self) -> None:
+        """Set last toggle direction when writing ha state."""
+        super().async_write_ha_state()
+        if self.state == STATE_OPENING:
+            self._is_last_toggle_direction_open = True
+        elif self.state == STATE_CLOSING:
+            self._is_last_toggle_direction_open = False
         else:
-            await self.async_close_cover_tilt(**kwargs)
+            # do nothing on all other states
+            pass
```

Comment:
No need for a default case, but you can keep the comment for clarity.

Reply:
Done ... tests are working!